docs: add markdown documentation (#4314)
vaadin/docs: articles/components/markdown/index.adoc

---
title: Markdown
page-title: Markdown component | Vaadin components
description: Markdown allows you to render Markdown content in your application with support for basic formatting, lists, links, and more.
meta-description: Use the Vaadin Markdown component to display Markdown formatted content in Vaadin applications.
page-links:
  - 'API: https://cdn.vaadin.com/vaadin-web-components/{moduleNpmVersion:@vaadin/markdown}/#/elements/vaadin-markdown[TypeScript] / https://vaadin.com/api/platform/{moduleMavenVersion:com.vaadin:vaadin}/com/vaadin/flow/component/markdown/Markdown.html[Java]'
  - 'Source: https://github.com/vaadin/web-components/tree/v{moduleNpmVersion:@vaadin/markdown}/packages/markdown[TypeScript] / https://github.com/vaadin/flow-components/tree/{moduleMavenVersion:com.vaadin:vaadin}/vaadin-markdown-flow-parent[Java]'
version: since:com.vaadin:vaadin@V24.8
---
= [since:com.vaadin:vaadin@V24.8]#Markdown#

// tag::description[]

> Markdown component allows the user to display formatted text content using the Markdown syntax.

Reviewer: ```suggestion
The Markdown component renders content written in the Markdown syntax as HTML.
```
